nixos/modules/services/hardware/nvidia-container-toolkit/default.nix

```diff
@@ -78,6 +78,11 @@
   };
 
   config = lib.mkIf config.hardware.nvidia-container-toolkit.enable {
+    assertions = [
+      { assertion = !config.hardware.nvidia.open;
+        message = "nvidia-container-toolkit requires the unfree Nvidia driver (set hardware.nvidia.open = false)";
```

Comment:
Does it though?

What about this alternative: we remove `videoDrivers = [ "nvidia" ]` and instead add `assertion = hardware.nvidia.datacenter.enable || elem "nvidia" videoDrivers`

Reply:
> Does it though?

Asking because surely it doesn't, now that nvidia is even deprecating the closed-source `nvidia_x11` package. It might, however, require `"nvidia"` over `"nouveau"` or something

Reply:
Yes, this is a better solution. Let me update the PR.

Reply:
My reading of nixos/modules/hardware/video/nvidia.nix is that currently the upstream open-source driver is enabled by

```
{
  ...videoDrivers = [ "nvidia" ];
  ...nvidia.open = true;
}
```

Which, btw, is implement in a truly horrific way (`if open then package.open else package.bin`, even though `package.open` and `package` are entirely different derivations and only one of them is source-built... boi does this need a refactoring